linode/manager
# feat: [M3-8103] - Refactor Image Create and Add Tags

## Description 📝

Refactors the Image Create page and adds the ability to add tags during creation. This PR does not touch the upload page. **These changes will not be feature flagged**.

## Changes  🔄
- Completely refactored the Image Create page to use `react-hook-form` ✏️ 
- Added tags input to the Image Create flow 🏷️ 
- Implements new UI and copy according the Figma mockup 🎨 
- Improves cypress tests for Image creation 🧪 
  - I added one spec that is 100% mocked data and another spect that is 100% real API requests (cc @jdamore-linode)

## Preview 📷

| Before  | After   |
| ------- | ------- |
| ![Screenshot 2024-05-14 at 7 52 30 PM](https://github.com/linode/manager/assets/115251059/c55ae3b3-bc5b-4621-aea4-a5c4c9ac89c7) | ![Screenshot 2024-05-14 at 7 52 42 PM](https://github.com/linode/manager/assets/115251059/a017dc38-13fb-4d6b-953f-4bc38856d2ff) |

## How to test 🧪

- Test creating an Image 💾 
- Using the "Network" tab of dev tools, verify the payload looks correct (includes tags) 🌏 
- Test validation on the Image Create form ☑️ 
- Test the Image Create page as a restricted user 👤 
- Verify there are no breaking changes ⛓️‍💥

## As an Author I have considered 🤔
